### ISSUE TITLE
`ana.int.refinement` can worsen precision

### ISSUE THREAD

**Issue by sim642** [goblint/analyzer#866]
Here's a minimized case, where @karoliineh used our `very-precise` example conf, which actually made things less precise.

```c
int main() {
  int i, j;
  if (i < j)
    return 1;
  else
    return 2;
}
```

Without int refinement, no warnings are printed:
```console
./goblint --enable ana.int.interval --set ana.int.refinement never
```

With int refinement, a spurious overflow warning is printed at the comparison:
```console
./goblint --enable ana.int.interval --set ana.int.refinement once
```
(Same with fixpoint refinement.)
